### PR TITLE
Update examples that start with "f" or "g" to use jest

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: examples/generics/test/test.ts [hunk 1]
```diff
@@ -1,13 +1,16 @@
 import { getCanisterId } from 'azle/dfx';
-import { runTests } from 'azle/test';
+import { runTests } from 'azle/test/jest';
 
 import { createActor } from './dfx_generated/generics';
 import { getTests } from './tests';
 
-const genericsCanister = createActor(getCanisterId('generics'), {
+// TODO is it time to get rid of this example?
```
Comment (bdemann) [inline examples/generics/test/test.ts:7]: @lastmjs is it?

File: examples/fetch_ic/test/tests.ts [hunk 2]
```diff
@@ -6,281 +6,221 @@ import * as dns from 'node:dns';
...
+            mainPage = pages[1];
+        });
 
-                await mainPage.waitForNetworkIdle();
+        wait('for identity to be set', 5_000);
 
-                const success = await page.evaluate(
-                    () => (window as any).headersObjectSuccess
-                );
+        it('Headers Array', async () => {
```
Comment (bdemann) [inline examples/fetch_ic/test/tests.ts:43]: @lastmjs I'll need some help with the names for these test

File: examples/file_protocol/test/tests.ts [hunk 3]
```diff
@@ -1,99 +1,56 @@
...
-        {
-            name: '/read-test0',
-            test: async () => {
-                try {
-                    const response = await fetch(
-                        `${origin}/read-test0?filename=${'assets/test0.txt'}`
-                    );
-                    const responseText = await response.text();
+    return () => {
+        it('/reads a file in the top level directory', async () => {
```
Comment (bdemann) [inline examples/file_protocol/test/tests.ts:10]: 'reads a file in the top level directory'

File: examples/generics/test/tests.ts [hunk 4]
```diff
@@ -1,347 +1,245 @@
...
-                        'Arm3' in result &&
-                        Array.isArray(result.Arm3) &&
-                        result.Arm3[0] === 'It did work'
-                };
-            }
-        }
-    ];
+export function getTests(genericsCanister: ActorSubclass<_SERVICE>): Test {
+    return () => {
+        it('simpleResult', async () => {
```
Comment (bdemann) [inline examples/generics/test/tests.ts:8]: If we decide to move forward with generic do we want to take the time to try and update it a bit and actually have it run in out tests suite? As part of that do we want to make these names more descriptive?

File: examples/guard_functions/test/tests.ts [hunk 5]
```diff
@@ -1,149 +1,89 @@
...
-                        )
-                    };
-                }
-            }
-        }
-    ];
+): Test {
+    return () => {
+        it('successfully calls an unguarded method', async () => {
+            const result = await guardFunctionsCanister.identifierAnnotation();
```
Comment (bdemann) [inline examples/guard_functions/test/tests.ts:11]: Were these from when we had a different API for guard methods? I don't exactly understand what they are doing

File: examples/generics/test/test.ts (hunk 1 above)
Comment (lastmjs) [reply examples/generics/test/test.ts]: Yes it is

File: examples/fetch_ic/test/tests.ts (hunk 2 above)
Comment (lastmjs) [reply examples/fetch_ic/test/tests.ts]: Yes we need to work on the names now based on the frontend tests

**Review by lastmjs — CHANGES_REQUESTED** [review]

**Pull request merged**